microcosm_daemon/health_reporter.py

```diff
@@ -0,0 +1,57 @@
+import os
+from logging import getLogger
+
+from microcosm.api import defaults, typed
+
+from microcosm_daemon.error_policy import ExitError
+
+
+try:
+    import requests
+except ImportError:
+    requests = None  # type:ignore
+
+
+logger = getLogger("daemon.health_reporter")
+
+
+class HealthReporter:
+    def __init__(self, graph):
+        self.healthcheck_server_host = graph.config.health_reporter.healthcheck_server_host
+        self.healthcheck_server_port = graph.config.health_reporter.healthcheck_server_port
+
+    def __call__(self, health, prev_health, errors):
+        self.heartbeat()
+
+        message = f"Health is {health}"
+
+        if prev_health != health:
+            logger.info(message)
+        else:
+            logger.debug(message)
+
+        for error in errors:
+            if isinstance(error, ExitError):
+                continue
+
+            logger.warn(f"Caught error during state evaluation: {error}", exc_info=True)
+
+    def heartbeat(self):
+        if requests is None:
+            return
+
+        try:
+            requests.post(
+                f"{self.healthcheck_server_host}:{self.healthcheck_server_port}/api/v1/heartbeat",
+                json={"pid": os.getpid()},
+            )
```

Comment:
Should these requests timeout?

Reply:
I would say it doesn't matter much. If this request does time out then it means the server on the main process isn't responding, which means it wouldn't be responding to healthchecks either, and the whole thing will be terminated.